server/src/cycles.ts

```diff
@@ -44,50 +44,61 @@ export function getCommandUriToOpenDocu(id: string | undefined): string {
  * @returns a {@link Cycle} object 
  */
 function jsonCycleToCycle(cycle: any): Cycle {
-    const parameterList: Array<Parameter> = cycle.ParameterList.map((parameter: any) => jsonParameterToParameter(parameter, cycle.DocumentationReference?.Parameter));
-    // sort params by name
-    parameterList.sort((a: Parameter, b: Parameter) => {
-        const numA = parseInt(a.name.replace(/\D/g, ''), 10);
-        const numB = parseInt(b.name.replace(/\D/g, ''), 10);
-        return numA - numB;
-    });
+    try {
+        const parameterList: Array<Parameter> = cycle.ParameterList.map((parameter: any) => jsonParameterToParameter(parameter, cycle.DocumentationReference?.Parameter));
+        // sort params by name
+        parameterList.sort((a: Parameter, b: Parameter) => {
+            const numA = parseInt(a.name.replace(/\D/g, ''), 10);
+            const numB = parseInt(b.name.replace(/\D/g, ''), 10);
+            return numA - numB;
+        });
 
-    const documentationReference = cycle.DocumentationReference ? new DocumentationReference(cycle.DocumentationReference.Overview, cycle.DocumentationReference.Parameter) : undefined;
-    const descriptionDictionary = new DescriptionDictionary(cycle.DescriptionDictionary["en-US"], cycle.DescriptionDictionary["de-DE"]);
-    return new Cycle(
-        cycle.Name,
-        cycle.Media,
-        documentationReference,
-        descriptionDictionary,
-        parameterList
-    );
+        const documentationReference = (cycle.DocumentationReference && cycle.DocumentationReference.Overview && cycle.DocumentationReference.Parameter) ? new DocumentationReference(cycle.DocumentationReference.Overview, cycle.DocumentationReference.Parameter) : undefined;
+        const descriptionDictionary = new DescriptionDictionary(cycle.DescriptionDictionary["en-US"], cycle.DescriptionDictionary["de-DE"]);
 
+        return new Cycle(
+            cycle.Name,
+            cycle.Media,
+            documentationReference,
+            descriptionDictionary,
+            parameterList
+        );
+    } catch (error) {
+        throw new Error("Failed to parse cycle " + cycle.Name + ": \n" + error);
```

Comment:
Was genau meinst du mit parsen des Zyklus?

Reply:
Parsen der JSON Infos, also ob in der JSON ein ungewohntes Verhalten ist